### PR TITLE
Introduce API dashboard and enhance security configuration

### PR DESCRIPTION
**Pull request by Pawel-Mackiewicz**
## Summary by CodeRabbit

- **New Features**
  - Introduced an API endpoint to retrieve the working balance of a user's account, reflecting available funds after accounting for pending transactions.
  - Added a response format that includes the working balance, account ID, and timestamp.
- **Bug Fixes**
  - Improved error handling for invalid API request parameters with clearer validation error responses.
- **Documentation**
  - Enhanced API documentation and test service Javadoc for better clarity.
- **Tests**
  - Added comprehensive integration tests for working balance calculations, including edge cases and concurrent scenarios.

### CODE REVIEW

File: src/main/java/info/mackiewicz/bankapp/system/error/handling/service/ValidationErrorProcessor.java [hunk 1]
```diff
@@ -68,6 +68,28 @@ public List<ValidationError> extractValidationErrors(ConstraintViolationExceptio
...
+        String causeMsg = ex.getCause().getMessage();
+        int startIndex = causeMsg.indexOf("\"") + 1;
+        int endIndex = causeMsg.lastIndexOf("\"");
+        String rejectedValue = causeMsg.substring(startIndex, endIndex);
+
+        return List.of(new ValidationError(
+                ex.getName(),
+                ex.getMessage(),
+                rejectedValue));
+    }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/main/java/info/mackiewicz/bankapp/system/error/handling/service/ValidationErrorProcessor.java]: done.

File: src/main/java/info/mackiewicz/bankapp/presentation/api/dashboard/service/ApiDashboardService.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+     */
+    public BigDecimal getWorkingBalance(int accountId) {
+        BigDecimal balance = accountRepository.findBalanceById(accountId)
+                .orElseThrow(
+                        () -> new AccountNotFoundByIdException("Account with ID " + accountId + " not found.")
+                );
+        BigDecimal balanceOnHold = transactionRepository.findBalanceOnHoldBySourceAccount_Id(accountId);
+
+        return balance.subtract(balanceOnHold);
+    }
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/main/java/info/mackiewicz/bankapp/presentation/api/dashboard/service/ApiDashboardService.java]: done.

File: src/main/java/info/mackiewicz/bankapp/core/transaction/repository/TransactionRepository.java [hunk 3]
```diff
@@ -36,6 +41,26 @@ Optional<List<Transaction>> findTopNByAccountIdOrderByCreatedDesc(
...
+        *
+        * @return The total amount on hold as a BigDecimal, or 0 if no matching transactions are found.
+        */
+       @Query("""
+                   SELECT COALESCE(SUM(t.amount), 0) 
+                          FROM Transaction t 
+                          WHERE t.sourceAccount.id = :accountId 
+                          AND t.status IN ('NEW', 'PENDING')
+               """)
+       BigDecimal findBalanceOnHoldBySourceAccount_Id(@Param("accountId") Integer accountId);
```
[the remark this answers is not in the document]
Comment (Pawel-Mackiewicz) [reply src/main/java/info/mackiewicz/bankapp/core/transaction/repository/TransactionRepository.java:65]: n/n

**Comment by Pawel-Mackiewicz** [comment]
@coderabbitai 
write `squash and merge` commit msg for this PR.
